Fixed the issue with formatting issues not being reported when verification fails

wasylkowski opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:
1. If this is your first time, read our contributor guidelines https://git.k8s.io/community/contributors/devel/pull-requests.md#the-pr-submit-process and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/devel/pull-requests.md#best-practices-for-faster-reviews
3. Follow the instructions for writing a release note: https://git.k8s.io/community/contributors/devel/pull-requests.md#write-release-notes-if-needed
-->

**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:
<!--  Steps to write your release note:
1. Use the release-note-* labels to set the release note state (if you have access)
2. Enter your extended release note in the below block; leaving it blank means using the PR title as the release note. If no release note is required, just write `NONE`.
-->
```release-note
NONE
```

wasylkowski-a committed:
Fixed the issue with formatting issues not being reported when verification fails.

wasylkowski commented:
/assign @fejta

eparis commented:
This does not appear to be a problem on my system. What problem is this supposed to fix?
```
$ go version
go version go1.8.1 linux/amd64

$ hack/verify-gofmt.sh 
diff ./cmd/genman/gen_kube_man.go gofmt/./cmd/genman/gen_kube_man.go
--- /tmp/gofmt337461115	2017-06-12 09:13:10.817350914 -0400
+++ /tmp/gofmt682044062	2017-06-12 09:13:10.817350914 -0400
@@ -38,7 +38,7 @@
 
 func main() {
 	// use os.Args instead of "flags" because "flags" will mess up the man pages!
-	  path := "docs/man/man1"
+	path := "docs/man/man1"
 	module := ""
 	if len(os.Args) == 3 {
 		path = os.Args[1]

$ echo $?
1

```

wasylkowski commented:
This is indeed very interesting, because after I fixed formatting issues I have had, and added a formatting issue given by you, I get the same (working) behavior without the fix. Let me look more deeply into why this is so

wasylkowski commented:
OK, I have found it. If the file happens not to parse correctly, verify_gofmt spits out a a not very useful message:
`
$ hack/verify-gofmt.sh
!!! Error in hack/verify-gofmt.sh:52
  Error in hack/verify-gofmt.sh:52. 'diff=$(find_files | xargs ${gofmt} -d -s 2>&1)' exited with status 123
Call stack:
  1: hack/verify-gofmt.sh:52 main(...)
Exiting with status 1
`

After my fix you get the following:
`
$ hack/verify-gofmt.sh
./test/e2e/stackdriver_monitoring.go:95:100: missing ',' in argument list
`

I believe the fix makes verify_gofmt more useful, but it is up to you to agree/disagree with me. Let me know what you think.

wasylkowski commented:
Of course I will change the comment in the file.

wasylkowski commented:
Comment updated.

eparis commented on hack/verify-gofmt.sh:
```diff
@@ -46,9 +46,12 @@ find_files() {
         -o -wholename '*/bindata.go' \
       \) -prune \
     \) -name '*.go'
-}
```
Where did this go?

wasylkowski replied:
Oops. Sorry about that. Brought it back.

cblecker commented:
With these changes, if the script is unable to parse the file and this new error is displayed, does the whole script itself still exit non-zero?

eparis commented:
It does. I double checked.

cblecker commented:
@k8s-bot ok to test
/release-note-none

eparis commented:
/lgtm

although I think this is going to have to wait for the queue to reopen for 1.8.

wasylkowski commented:
Thanks Eric!

eparis commented:
/approve no-issue

fejta commented:
/unassign

mwielgus commented:
@k8s-bot test this please
